Nodejs/Product/Nodejs/Debugger/NodeDebugger.cs

```diff
@@ -788,14 +794,34 @@ public NodeDebugger(Uri debuggerEndpointUri, int id)
                 int line = stackFrame.Line;
                 int column = stackFrame.Column;
                 string functionName = stackFrame.FunctionName;
+                int lineEnd = stackFrame.EndLine;
+
+                string name = "";
 
                 // Map file position to original, if required
                 if (module.JavaScriptFileName != module.FileName) {
-                    SourceMapInfo mapping = SourceMapper.MapToOriginal(module.JavaScriptFileName, line, column);
-                    if (mapping != null) {
-                        line = mapping.Line;
-                        column = mapping.Column;
-                        functionName = string.IsNullOrEmpty(mapping.Name) ? functionName : mapping.Name;
+                    DecodedSourceMap decodedSourceMap = null;
+                    DkmTextSpan? tsSpan = null;
+                    string jsFileName = module.JavaScriptFileName;
+                    string sourceMapFilename = SourceMapFileHelper.FindSourceMapFile(jsFileName);
+                    SourceMapSourceInfo sourceInfo = null;
+                    if (!string.IsNullOrEmpty(sourceMapFilename)) {
+
+                        if (DecodedSourceMaps.ContainsKey(sourceMapFilename)) {
+                            decodedSourceMap = DecodedSourceMaps[sourceMapFilename];
+                        } else {
+                            decodedSourceMap = SourceMapReader.LoadSourceMap(jsFileName, sourceMapFilename);
+                            DecodedSourceMaps.Add(sourceMapFilename, decodedSourceMap);
+                            _sourceMapFiles.Add(jsFileName, sourceMapFilename);
+                        }
+
+                        tsSpan = decodedSourceMap.MapJsSourcePosition(new DkmTextSpan(line + 1, line + 1, column + 1, column + 1), out sourceInfo, out name);
```

Comment:
[nit] The logic for creating calling `MapJsSourcePosition` and creating `DkmTextSpan` from the line and column by adding 1s is repeated a few times. Extract to a helper function to prevent future errors such as forgetting to add the 1s on line or column.